internal/monitor/monitor.go

```diff
@@ -88,17 +98,27 @@ func (pm *PowerMonitor) Init() error {
 func (pm *PowerMonitor) signalNewData() {
 	select {
 	case pm.dataCh <- struct{}{}: // send signal to any waiting goroutinel
+		pm.logger.Debug("Data channel updated")
 	default:
+		pm.logger.Debug("Data channel is full, dropping signal")
 	}
 }
 
 func (pm *PowerMonitor) Run(ctx context.Context) error {
 	pm.logger.Info("Monitor is running...")
+	go pm.collectionLoop() // NOTE: runs in background
```

Comment:
there is no need to run this in separate go routine, its lifetime is short and is not a loop.

Reply:
its lifetime will be long as soon as we add the process and containers and pods - right?

Reply:
this function does first collection, schedules timer, and returns.